### DISCUSSION TITLE
Combining points and polygons - in package or an add-on package?

### DISCUSSION THREAD

**Discussion by Robinlovelace** [ropensci/osmextract#215] [Ideas]
Sometimes it's useful to combine polygons and points. For example, supermarkets are represented as both points and polygons. I've written a test function that combines them: https://github.com/Nowosad/SIGR2021/blob/master/workshop1/osm_combine.R

Think this is worth packaging but, in the interests of keeping osmextract small and maintainable would suggest putting this general osm processing functionality in separate package, e.g. called `osmextras`.

Thoughts?

**Comment by agila5** [comment]
Hi @Robinlovelace! Are you sure that the code written in that function is appropriate for `osmextract`? I think that is more relevant for `osmdata` since, AFAIK, the "duplicated" points are "removed" when building the POLYGON objects. 

For example: 

``` r
# package
library(osmextract)
#> Data (c) OpenStreetMap contributors, ODbL 1.0. https://www.openstreetmap.org/copyright.
#> Check the package website, https://docs.ropensci.org/osmextract/, for more details.
library(sf)
#> Linking to GEOS 3.9.0, GDAL 3.2.1, PROJ 7.2.1

# get supermarks
supermarkets_points <- oe_get(
  "Isle of Wight", 
  layer = "points", 
  extra_tags = "shop", 
  query = "SELECT * FROM points WHERE shop = 'supermarket'"
)
#> The input place was matched with: Isle of Wight
#> The chosen file was already detected in the download directory. Skip downloading.
#> The corresponding gpkg file was already detected. Skip vectortranslate operations.
#> Reading layer `points' from data source 
#>   `C:\Users\Utente\Documents\osm-data\geofabrik_isle-of-wight-latest.gpkg' 
#>   using driver `GPKG'
#> Simple feature collection with 7 features and 11 fields
#> Geometry type: POINT
#> Dimension:     XY
#> Bounding box:  xmin: -1.29727 ymin: 50.58849 xmax: -1.145604 ymax: 50.76113
#> Geodetic CRS:  WGS 84
supermarkets_polygons <- oe_get(
  "Isle of Wight", 
  layer = "multipolygons", 
  extra_tags = "multipolygons", 
  query = "SELECT * FROM multipolygons WHERE shop = 'supermarket'"
)
#> The input place was matched with: Isle of Wight
#> The chosen file was already detected in the download directory. Skip downloading.
#> The corresponding gpkg file was already detected. Skip vectortranslate operations.
#> Reading layer `multipolygons' from data source 
#>   `C:\Users\Utente\Documents\osm-data\geofabrik_isle-of-wight-latest.gpkg' 
#>   using driver `GPKG'
#> Simple feature collection with 18 features and 26 fields
#> Geometry type: MULTIPOLYGON
#> Dimension:     XY
#> Bounding box:  xmin: -1.515673 ymin: 50.63308 xmax: -1.142295 ymax: 50.76222
#> Geodetic CRS:  WGS 84

# intersection
lengths(st_intersects(supermarkets_points, supermarkets_polygons))
#> [1] 0 0 0 0 0 0 0
```

<sup>Created on 2021-06-30 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

**Reply by Robinlovelace** [reply]
> Are you sure that the code written in that function is appropriate for `osmextract`?

No and agree, `osmextract` would be be a better home for this. However... I think there are some instances where both point and polygon representations of supermarkets and other features exist.

**Reply by Robinlovelace** [reply]
See here for a reproducible example which did have some duplicate points, at least in early 2021, the idea was developed in the context of data from `osmextract`: https://github.com/cyipt/actdev/blob/main/code/get_pois.R
